src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,29 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getAge() > 17 && user.getPassword() != null
+                && user.getPassword().length() > 5 && !findMatchLogin(user)) {
+            storageDao.add(user);
+            return user;
+        }
+        throw new RuntimeException("Input Errors");
+    }
+
+    private boolean findMatchLogin(User user) {
+        for (User current: Storage.people) {
+            if (current.getLogin().equals(user.getLogin())) {
+                return true;
+            }
+
+        }
+        return false;
```

Comment:
lets use get(String login) method from StorageDaoImpl to check if a user with this email exists